libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -755,7 +759,11 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
       for (const [importerPackageName, importerVersionSpecifier] of Object.entries(specifiers)) {
         const foundDependency: PackageJsonDependency | undefined =
           dependencyVersions.get(importerPackageName);
-        if (!foundDependency || foundDependency.version !== importerVersionSpecifier) {
+        if (!foundDependency) {
+          return true;
+        }
+        const resolvedVersion: string = this.overrides.get(importerPackageName) ?? foundDependency.version;
```

Comment:
The `overrides` field also accepts keys like `@rushstack/heft@<=0.50.6`, or `@rushstack/heft>@rushstack/node-core-library`, not just package names, but I guess this will catch the basic case.